src/tools/kdb/ls.cpp

```diff
@@ -21,12 +21,9 @@ LsCommand::LsCommand () : kdb (root)
 {
 }
 
-int LsCommand::execute (Cmdline const & cl)
+int LsCommand::execute (const Cmdline & cl)
```

Comment:
Ok i have to admit i haven't thought about the ref counter of the Keys yet. You are right in that cases it probably makes more sense to pass them around by reference if applicable to avoid that.

But i meant the general cases where i pass a const object around, for example a string. Should i use `const string foo` or `string const foo` in the future?

Doesn't the const int indicate that this integer will not change in this method? I've seen it more from the perspective of immutability. If i want to express that some parameter should not change in a method, i mark it as const, regardless of wheter its getting passed by copy so modifications wouldn't make a difference outside the method.